File: extract_pdf.js
// Extract text from the vocabulary PDF using pdf-parse v2
const fs = require('fs');
const { PDFParse } = require('pdf-parse');

const pdfPath = 'E:\\Tina\\中考考纲\\003-无水印 2026高清中考词汇小绿本+赠送+配套音频\\无水印高清2026中考词汇小绿本3本\\1. 26年初中英语考纲词汇用法手册（最新版）370页.pdf';

async function main() {
  const dataBuffer = fs.readFileSync(pdfPath);
  console.log('PDF size:', dataBuffer.length, 'bytes');
  
  // Convert Buffer to Uint8Array
  const uint8 = new Uint8Array(dataBuffer);
  const parser = new PDFParse(uint8);
  await parser.load();
  
  const info = await parser.getInfo();
  console.log('Info:', JSON.stringify(info).substring(0, 500));
  
  // Extract text page by page
  let allText = '';
  const numPages = info.pages || 370;
  
  for (let i = 1; i <= numPages; i++) {
    try {
      const pageText = await parser.getText(i);
      allText += `\n\n---PAGE ${i}---\n\n` + pageText;
      if (i % 50 === 0) console.log('Processed page', i);
    } catch(e) {
      console.log('Error on page', i, e.message);
    }
  }
  
  fs.writeFileSync('E:\\Tina\\自研背单词软件\\raw_text.txt', allText, 'utf-8');
  console.log('Total text length:', allText.length);
  console.log('First 3000 chars:');
  console.log(allText.substring(0, 3000));
  
  parser.destroy();
}

main().catch(err => console.error(err));

File: extract_pdf.py
import pdfplumber
import json
import sys

pdf_path = r'E:\Tina\中考考纲\003-无水印 2026高清中考词汇小绿本+赠送+配套音频\9. 中考英语考纲词汇（详细版）.pdf'
output_path = r'E:\Tina\自研背单词软件\pdf_full_text.txt'

with pdfplumber.open(pdf_path) as pdf:
    print(f"Total pages: {len(pdf.pages)}")
    all_text = []
    for i, page in enumerate(pdf.pages):
        text = page.extract_text()
        if text:
            all_text.append(f"=== Page {i+1} ===\n{text}")
        if (i+1) % 10 == 0:
            print(f"Processed {i+1}/{len(pdf.pages)} pages")
    
    full_text = "\n\n".join(all_text)
    with open(output_path, 'w', encoding='utf-8') as f:
        f.write(full_text)
    print(f"Extracted {len(full_text)} chars to {output_path}")
    
    # Search for adventure
    idx = full_text.lower().find('adventure')
    if idx >= 0:
        print(f"\nFound 'adventure' at index {idx}:")
        print(full_text[max(0,idx-100):idx+200])
    else:
        print("\n'adventure' not found in PDF text")
